engine/src/main/java/org/camunda/bpm/engine/impl/identity/db/DbIdentityServiceProvider.java

```diff
@@ -299,7 +299,7 @@ public IdentityOperationResult createMembership(String userId, String groupId) {
   public IdentityOperationResult deleteMembership(String userId, String groupId) {
     checkAuthorization(Permissions.DELETE, Resources.GROUP_MEMBERSHIP, groupId);
     if (existsMembership(userId, groupId)) {
-      deleteAuthorizations(Resources.GROUP_MEMBERSHIP, groupId);
+      deleteAuthorizationsForUser(Resources.GROUP_MEMBERSHIP, groupId, userId);
```

Comment:
📓 (just adding a note) I found out that we don't create authorizations when creating a membership:
https://github.com/camunda/camunda-bpm-platform/blob/204b40921f7749a916b28c1073e0dc8df5c27134/engine/src/main/java/org/camunda/bpm/engine/impl/cfg/auth/DefaultAuthorizationProvider.java#L104-L109
Still, I think the `#deleteAuthorizationsForUser()` must be part of the `#deleteMembership()`

Reply:
Yeah. I saw this as well. However, users might expect it to work like this, so I decided against removing it entirely. Also, API like assigning a tenant `IdentityService#deleteTenantUserMembership`, does something comparable.